### PR TITLE
[Feature] Card

### PR DESCRIPTION
**Pull request by Major-Eth**
Addresses https://github.com/Major-Eth/yearn-web-lib/issues/16
Create a basic `<Card />` component.

- [X] The component should be `<section>`
- [X] This should use the `colors.surface` colors from the theme
- [X] Paddings should be 24px
- [X] Rounded should be 8px
- [X] Hover style should be available with different style based on the theme. It should be disabled by default and available as optional prop.
- [X] On hover, transition should be smooth and cursor should be pointer

[Figma](https://www.figma.com/file/bFN3UFHtXuOwDMeml3GgKV/yearn-polished?node-id=1265%3A8769)

### CODE REVIEW

File: src/components/Card/Card.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  withHover?: boolean;
+}
+
+export const Card = ({ children, ...props }: CardProps) => (
+    <StyledCard {...props}>
+      {children}
+    </StyledCard>
+  );
+
+const StyledCard = styled.section`
```
Comment (karelianpie) [inline src/components/Card/Card.tsx:15]: nit: We can add the generic after the element and then we don't need to give the prop type on line 23, makes it a bit cleaner IMO

```suggestion
const StyledCard = styled.section<CardProps>`
```

Comment (Major-Eth) [reply src/components/Card/Card.tsx]: Fixed with [b2d44a](https://github.com/Major-Eth/yearn-web-lib/pull/20/commits/b2d44a7bbff0606b6cda65ba4a8b28e2263cb7f9) with linting

File: src/components/Card/Card.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+export interface CardProps {
+  children: ReactNode;
+  withHover?: boolean;
+}
+
+export const Card = ({ children, ...props }: CardProps) => (
+    <StyledCard {...props}>
+      {children}
+    </StyledCard>
+  );
```
[the remark this answers is not in the document]
Comment (Major-Eth) [reply src/components/Card/Card.tsx]: Fixed with [b2d44a](https://github.com/Major-Eth/yearn-web-lib/pull/20/commits/b2d44a7bbff0606b6cda65ba4a8b28e2263cb7f9) with linting

File: src/components/Card/Card.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+export const Card = ({ children, ...props }: CardProps): ReactElement => (
+  <StyledCard {...props}>{children}</StyledCard>
+);
+
+const StyledCard = styled.section<CardProps>`
+  background: ${({ theme }) => theme.colors.surface};
+  box-shadow: ${({ theme }) => `0px 0px 0px ${theme.colors.background}00`};
+  border-radius: 0.5rem;
```
Comment (karelianpie) [inline src/components/Card/Card.tsx:16]: Will this translate to 8px? I'm used to just put the 8px, wondering how you've come up with 0.5 rem :)

File: stories/components/Card.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  marginBottom: 0.5rem;
+`;
+const StyledText = styled(Text)`
+  color: ${({ theme }) => theme.colors.texts};
+`;
+
+const Template: ComponentStory<typeof CardComponent> = (props) => (
+  <CardComponent {...props}>
+    <StyledTtitle>This is my title</StyledTtitle>
+    <StyledText>{'There are many like it, but this one is yours. You can always return here for a bird\'s-eye view of your holdings. The cards below show the total balance and utilization of your wallet, and the holdings and performance for every Yearn product you use.'}</StyledText>
```
Comment (karelianpie) [inline stories/components/Card.stories.tsx:22]: nit: Worth to use ticks in order to not have to escape the single quote, I know this is just a story though

```suggestion
    <StyledText>{`There are many like it, but this one is yours. You can always return here for a bird's-eye view of your holdings. The cards below show the total balance and utilization of your wallet, and the holdings and performance for every Yearn product you use.`}</StyledText>
```

**Review by karelianpie — APPROVED** [review]
LGTM, very generic, but that's good for now

File: src/components/Card/Card.tsx (hunk 3 above)
Comment (Major-Eth) [reply src/components/Card/Card.tsx:16]: It's generally better to use `rem`, especially for compatibility with modern browser and IOS devices (`1rem = 16px`). Harder to get into it but produce good results

Comment (karelianpie) [reply src/components/Card/Card.tsx:16]: Nice! I'll keep this in mind and start using `rem` going forward

**Pull request merged**